### ISSUE TITLE
Add developer docs on Penumbra SDK

### ISSUE THREAD

**Issue by conorsch** [penumbra-zone/guide#34]
The current dev docs https://guide.penumbra.zone/dev don't sufficiently explain how to code against Penumbra, when writing integrations or applications. We should absolutely call out the fact that although Penumbra is a Cosmos chain, one cannot use the Cosmos SDK. This violation of assumptions has caused friction for several devs, so let's write it down. Consider the following feedback shared from a recent developer:

------------
* Penumbra documentation suggestions
  * RPC service
    * Create a document that outlines how to interact with the tendermint-compatible blockchain service using a unique transaction format based on actions instead of traditional messages.
    * Describe supported endpoints , like status, blocks, block_results, tx, and events subscriptions.
* Transaction payload
   * Transactions are structured around actions rather than messages. Explain the payload differences with examples.
* Actions
  * Describe supported actions like spend,output, undelegateA
  * Describe tx parameters, detection data, etc
* Events
  * List supported events, and show how clients can subscribe to various events emitted by the blockchain.
  * Describe supported events e.g.  EventBlockRoot,  EventAnchor , etc
-------------

In a nutshell, we need to highlight the importance of using custom protobuf messages via the buf tooling. That would go a long way in saving new devs time while orienting themselves to build against Penumbra.
